### ISSUE TITLE
Weight decay regularization

### ISSUE THREAD

**Issue by jSaso** [awslabs/djl#40]
## Description
Weight decay regularization besides, l1 and l2 regularization.

### Will this change the current api? How?
Another implementation of regularization.

### Who will benefit from this feature?
Everybody, learning will be faster.

## References
https://www.fast.ai/2018/07/02/adam-weight-decay/

**Comment by zachgk** [comment]
I think we have weight decay regularization. You can set it using `Optimizer.Builder.optWeightDecays(float weightDecays)` (see [javadoc](https://javadoc.djl.ai/api/0.3.0/ai/djl/training/optimizer/Optimizer.OptimizerBuilder.html#optWeightDecays-float-) or [source](https://github.com/awslabs/djl/blob/master/api/src/main/java/ai/djl/training/optimizer/Optimizer.java#L158)). I'll close it for now, but please reopen this if I'm wrong.
